tests/end2end/features/test_editor_bdd.py

```diff
@@ -180,18 +180,25 @@ def save_editor_wait(tmpdir):
     os.kill(pid, signal.SIGUSR2)
 
 
-@bdd.when(bdd.parsers.parse('I set up a fake "{kind}" fileselector '
-                            'selecting "{files}"'))
-def set_up_fileselector(quteproc, py_proc, kind, files):
+@bdd.when(bdd.parsers.parse('I setup a fake "{kind}" fileselector '
+                            'selecting "{files}" and writes to "{output_type}"'))
+def set_up_fileselector(quteproc, py_proc, kind, files, output_type):
     """Set up fileselect.xxx.command to select the file(s)."""
     cmd, args = py_proc(r"""
         import os
         import sys
-        tmp_file = sys.argv[1]
-        with open(tmp_file, 'w') as f:
-            for selected_file in sys.argv[2:]:
-                f.write(os.path.abspath(selected_file) + "\n")
+        if '--' in sys.argv:
```

Comment:
Why check for another marker here rather than checking for `{}`?

Reply:
Since when this gets run, the `{}` is already replaced with the path to the temporary file

Reply:
Ah, I see. I think it'd be good to make it a bit more obvious, e.g. by using `--file` instead. With my suggestion to add a test with `--file={}` as well, that'd then be consistent too.